common/components/componentsBySection/FindProjects/Filters/ProjectFilterDataContainer.jsx

```diff
@@ -0,0 +1,150 @@
+import RenderFilterCategory from "./RenderFilterCategory.jsx";
+import metrics from "../../../utils/metrics";
+import _ from 'lodash'
+
+/**
+ * @category: Tag category to pull from
+ * @title: Title of the dropdown
+ */
+type Props = {|
+  title: string
+|};
+
+type State = {|
+  tags: ?$ReadOnlyArray<TagDefinition>,
+  filterCounts: ?{ [key: string]: number },
+  selectedTags: ?{ [key: string]: boolean },
+|};
+
+class ProjectFilterDataContainer extends React.Component<Props, State> {
+  constructor(props: Props): void {
+    super(props);
+    this.state = {
+      tags: null,
+      selectedTags: null,
+      filterCounts: null
+    };
+
+    // passing true to fetchTagsByCategory asks backend to return num_times in API response
+    ProjectAPIUtils.fetchAllTags(true, tags => {
+      //Need to do some work before setting state: Remove empty tags, generate cat/subcat totals, cleanup, then set state.
+      //Remove tags from API with num_times=0 before doing anything else
+      let filteredTags = tags.filter(function(key) {
+          return key.num_times > 0;
+        })
+      //Generate category and subcategory totals - this is number of FILTERS and not total number of PROJECTS
+      //So this may be used for "Select All" checkbox reference but will not be used for display of counts in DOM
+      let subcatCount = _.countBy(filteredTags, 'subcategory' );
+      let catCount = _.countBy(filteredTags, 'category');
+      let countMergeResult = _.merge(catCount, subcatCount)
+      //remove unneeded count of empty category/subcategory entries
+      delete countMergeResult['']
+      //Group tags by category before generating child components
+      let sorted = _.groupBy(filteredTags, 'category');
+
+      //last, set state with our computed data
+      this.setState({
+        tags: filteredTags,
+        filterCounts: countMergeResult,
+        sortedTags: sorted
+      });
+    });
+    this._checkEnabled = this._checkEnabled.bind(this);
+    this._selectOption = this._selectOption.bind(this);
+  }
+
+  static getStores(): $ReadOnlyArray<FluxReduceStore> {
+    return [ProjectSearchStore];
+  }
+
+  static calculateState(prevState: State): State {
+    return {
+      selectedTags:_.mapKeys(ProjectSearchStore.getTags().toArray(), (tag: TagDefinition) => tag.tag_name)
+    };
+  }
+
+
+  render(): React$Node {
+    //should render a number of <RenderFilterCategory> child components
+    return (
+      <div>
+        { this.state.sortedTags ? this._renderFilterCategories() : null }
+      </div>
+    );
+  }
+
+  _renderFilterCategories(): void {
+    //iterate through this.state.sortedTags into key/value pairs, one per component
+    //first get category names in an array to iterate over
+
+
+    //define the order we want our filters to be in (these must match database "category" values to be valid)
+    const fixedOrderCategories = ["Issue(s) Addressed", "Role", "Technologies Used", "Project Stage", "Organization"]
```

Comment:
I think we might be able to simplify the sorting logic if we use a category configuration object.  For example:
const categoryConfig = {"Issue(s) Addressed":{orderNumber: 1},"Role": {orderNumber: 2} ...}

And then we can do the sorting by doing something like
const categoryGroups = _.toPairs(this.state.sortedTags);
const orderedFilters = _.sortBy(categoryGroups , categoryGroup => tag.category in this.state.sortedTags? categoryConfig[categoryGroup[0]] : 99);